utils/near-rate-limiter/src/framed_read.rs

```diff
@@ -413,4 +424,92 @@ mod tests {
         assert_eq!(throttle_controller.consume_bandwidth_used(), 1015);
         assert_eq!(throttle_controller.consume_bandwidth_used(), 0);
     }
+
```

Comment:
Unrelated to the PR, but rather than `ready!(PollSemaphore::poll_next(Pin::new(pinned.semaphore), cx));` we should do  `ready!(pinned.semaphore.poll_acquire(cx));` in the code above.